File: csharp/2014/transcendence-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Transcendence" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Transcendence",
                year: 2014,
                plot: "A scientist's drive for artificial intelligence, takes on dangerous implications when his consciousness is uploaded into one such program.",
                rating: 6.3
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Transcendence",
                year: 2014
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}

File: python/2014/transcendence-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Transcendence to the database
movies.insert(
    title="Transcendence", 
    year=2014, 
    plot="A scientist's drive for artificial intelligence, takes on dangerous implications when his consciousness is uploaded into one such program.", 
    rating=6.3
)

# Confirm that the movie was added
movie = movies.select(
    title="Transcendence", 
    year=2014
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    